src/skin/legacyskinparser.cpp

```diff
@@ -1084,6 +1089,29 @@ QWidget* LegacySkinParser::parseCoverArt(QDomElement node) {
     return pCoverArt;
 }
 
+void LegacySkinParser::parseSingletonDefinition(QDomElement node) {
+    // The actual singleton definition is identical to any other WWidgetGroup.
+    QWidget* child_group = parseWidgetGroup(node);
```

Comment:
probably should call ```child_group->hide()```. If no insertion point is visible to start with then the singleton will show up wherever it was defined in the skin tree.